tfhe/benches/integer/bench.rs

```diff
@@ -1400,6 +1404,121 @@ mod cuda {
         bench_group.finish()
     }
 
+    fn cuda_compress(c: &mut Criterion) {
+        let bench_name = "integer::cuda::compression";
+        let mut bench_group = c.benchmark_group(bench_name);
+        bench_group
+            .sample_size(15)
+            .measurement_time(std::time::Duration::from_secs(30));
+
+        let stream = CudaStreams::new_multi_gpu();
+
+        let param = PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+        let comp_param = COMP_PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M64;
+
+        let log_message_modulus = param.message_modulus.0.ilog2() as usize;
+
+        for num_bits in [
+            8,
+            16,
+            32,
+            64,
+            128,
+            256,
+            comp_param.lwe_per_glwe.0 * log_message_modulus,
+        ] {
+            assert_eq!(num_bits % log_message_modulus, 0);
+            let num_blocks = num_bits / log_message_modulus;
+
+            // Generate private compression key
+            let (cks, _) = KEY_CACHE.get_from_params(param, IntegerKeyKind::Radix);
+            let private_compression_key = cks.new_compression_private_key(comp_param);
+
+            // Generate and convert compression keys
+            let (radix_cks, _) = gen_keys_radix_gpu(param, num_blocks, &stream);
+            let (compressed_compression_key, _) =
+                radix_cks.new_compressed_compression_decompression_keys(&private_compression_key);
+            let cuda_compression_key = compressed_compression_key.decompress_to_cuda(&stream);
+
+            // Encrypt
+            let ct = cks.encrypt_radix(0_u32, num_blocks);
+            let d_ct = CudaUnsignedRadixCiphertext::from_radix_ciphertext(&ct, &stream);
+
+            // Benchmark
+            let mut builder = CudaCompressedCiphertextListBuilder::new();
+
+            builder.push(d_ct, &stream);
+
+            bench_group.bench_function(format!("compress_u{num_bits}"), |b| {
+                b.iter(|| {
+                    let compressed = builder.build(&cuda_compression_key, &stream);
+
+                    _ = black_box(compressed);
+                })
+            });
+        }
+    }
+
+    fn cuda_decompress(c: &mut Criterion) {
+        let bench_name = "integer::cuda::compression";
+        let mut bench_group = c.benchmark_group(bench_name);
+        bench_group
+            .sample_size(15)
+            .measurement_time(std::time::Duration::from_secs(30));
+
+        let stream = CudaStreams::new_multi_gpu();
+
+        let param = PARAM_MESSAGE_2_CARRY_2_KS_PBS;
```

Comment:
params don't match the compression params, if we have the same error for the CPU it needs a fix we can do separately